pkg/storage/ingest/partition_offset_reader.go

```diff
@@ -187,9 +241,7 @@ func (p *partitionOffsetReader) FetchLastProducedOffset(ctx context.Context) (_
 		return 0, err
 	}
 
-	// The offset we get is the offset at which the next message will be written, so to get the last produced offset
-	// we have to subtract 1. See DESIGN.md for more details.
-	return listRes.Topics[0].Partitions[0].Offset - 1, nil
+	return listRes.Topics[0].Partitions[0].Offset, nil
```

Comment:
Note to reviewers: the `- 1` has been moved to `FetchLastProducedOffset()` because we only need to apply when querying "partition end offset" to get the "last produced offset" (because "partition end offset" is the offset at which the **next** record will be published to).